internal/gateway/frontend/leases.go

```diff
@@ -124,12 +124,13 @@ func handleCommitLease(services be.ActionController, token string, w http.Respon
 	}
 
 	msg := make(map[string]interface{})
-	if err := services.CommitLease(
+	if finalRev, err := services.CommitLease(
 		ctx, token, reqMsg.OldRootHash, reqMsg.NewRootHash, reqMsg.RepositoryTag); err != nil {
 		msg["status"] = "error"
 		msg["reason"] = err.Error()
 	} else {
 		msg["status"] = "ok"
+		msg["final_revision"] = finalRev
```

Comment:
Does the gateway code handle this right? Or at least it doesn't explode?

Reply:
This is solved by https://github.com/cvmfs/cvmfs/pull/2508/files in the publisher code. It only works from 2.7.3 onwards, but Jakob said that this is fine.